### PR TITLE
feat: add notification and update deps

### PR DESCRIPTION
**Pull request by zongUMR**
## What has been implemented?

Closes https://app.asana.com/0/1200011502754281/1200140904325774/f

1. Update `serverless/utils` to 4.x
2. Add **questionnaire** notification for the above task

## Steps to verify

- Run X
- ...

## Todos:

- [ ] Write tests
- [ ] Write / update documentation
- [ ] Run Prettier

### CODE REVIEW

File: src/cli/commands-cn/run.js [hunk 1]
```diff
@@ -107,7 +107,10 @@ module.exports = async (config, cli, command) => {
   let deferredNotificationsData;
   if (command === 'deploy') {
     deferredNotificationsData = requestNotification(
-      Object.assign(generateNotificationsPayload(instanceYaml), { command: 'deploy' })
+      Object.assign(generateNotificationsPayload(instanceYaml), {
+        command: 'deploy',
+        actions: [utils.notificationActions.components_questionnaire],
```
Comment (medikoo) [inline src/cli/commands-cn/run.js:112]: This looks as a static property, why exactly it's added?

**Review by medikoo — COMMENTED** [review]
Thanks @zongUMR ! I have just one question and we should be good

File: src/cli/commands-cn/run.js (hunk 1 above)
Comment (zongUMR) [reply src/cli/commands-cn/run.js:112]: The backend service will use actions to insert different notify messages, you can the the backedn PR here: https://github.com/serverlessinc/platform-cn/pull/165/files

**Review by medikoo — COMMENTED** [review]
> The backend service will use actions to insert different notify messages, you can see the the backend PR here: https://github.com/serverlessinc/platform-cn/pull/165/files

Thanks for explanations, still it looks as if it try to go against the design of backend notification service.

The way it is designed, is that depending on the payload data (which describes deployed service characteristics, and provides basic installation info), it's _backend_ service that decides which notification to show. e.g. if backend will notice that payload reflects outdated version, it returns version update notification.

While what you proposed here is that client instead of sending service characteristics in a payload, asks for specific notification. Is my understanding correct?

What's the point of sending such information from client?

**Comment by zongUMR** [comment]
> > The backend service will use actions to insert different notify messages, you can see the the backend PR here: https://github.com/serverlessinc/platform-cn/pull/165/files
> 
> Thanks for explanations, still it looks as if it try to go against the design of backend notification service.
> 
> The way it is designed, is that depending on the payload data (which describes deployed service characteristics, and provides basic installation info), it's _backend_ service that decides which notification to show. e.g. if backend will notice that payload reflects outdated version, it returns version update notification.
> 
> While what you proposed here is that client instead of sending service characteristics in a payload, asks for specific notification. Is my understanding correct?
> 
> What's the point of sending such information from client?

Got it, this is only a static notify, it does not need to send anything to backend. I need to only show the notification in CLI. A new commit fix it. Thanks @medikoo

File: src/cli/commands-cn/run.js [hunk 2]
```diff
@@ -164,7 +167,15 @@ module.exports = async (config, cli, command) => {
   }
   cli.sessionStop('success', 'Success');
 
-  if (deferredNotificationsData) printNotification(cli, await deferredNotificationsData);
+  if (deferredNotificationsData) {
```
Comment (timqian) [inline src/cli/commands-cn/run.js:170]: Looks like this static notification will only be shown when we got a backend notification. Is this what we want?

**Comment by ole3021** [comment]
@zongUMR  We should not implement sending the questionnaire logic on the client (CLI) side. 

if I'm understanding right, we have to trigger the server-side notification with a `NotificationsPayload`, but I'm not sure what's really the static notification is hrere. 

The ideal notification process is that

After the user make a deployment, and the server can send back a message to the user CLI console with the notification message that we like to show to the user. Also 
* This should work with back-compatible, no matter which version the user is using should get the messages.
* We can controller easier to add, change, remove the messages, without change anything on the CLI side.
    * so that we can add a new message, and remove messages as we want. CLI doesn't need to know about that.

Also from our design principle: We should keep our CLI simple clean and not include any business logic, all the logic should be not related to a specific version of CLI but implement on the backend side as much as possible.

File: src/cli/commands-cn/run.js (hunk 2 above)
Comment (zongUMR) [reply src/cli/commands-cn/run.js:170]: This is the current behavior, we will not change it.

**Comment by zongUMR** [comment]
@medikoo @ole3021 @timqian This PR will not change any code, only update a package which includes the **China dev notify url** for testing: https://github.com/serverless/utils/pull/78, the real logic will put on server side

**Review by timqian — APPROVED** [review]

**Pull request merged**